misc/dbt-materialize/tests/adapter/test_get_last_relation_modified.py

```diff
@@ -0,0 +1,75 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License in the LICENSE file at the
+# root of this repository, or online at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import pytest
+
+from dbt.cli.main import dbtRunner
+
+
+freshness_via_metadata_schema_yml = """version: 2
+sources:
+  - name: test_source
+    freshness:
+      warn_after: {count: 10, period: hour}
+      error_after: {count: 1, period: day}
+    schema: "{{ env_var('DBT_GET_LAST_RELATION_TEST_SCHEMA') }}"
+    tables:
+      - name: test_table
+"""
```

Comment:
🚨🚨 We should **explicitly disallow the following (scary) scenario:** 🚨🚨

What happens if a user passes a `loaded_at_field` here? IIUC from [dbt Core #8704](https://github.com/dbt-labs/dbt-core/issues/8704), if this field is present, dbt will not use the system catalog to calculate freshness, but will instead invoke the [`collect_freshness`](https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/include/global_project/macros/adapters/freshness.sql) macro:

```sql
{% macro default__collect_freshness(source, loaded_at_field, filter) %}
  {% call statement('collect_freshness', fetch_result=True, auto_begin=False) -%}
    select
      max({{ loaded_at_field }}) as max_loaded_at,
      {{ current_timestamp() }} as snapshotted_at
    from {{ source }}
    {% if filter %}
    where {{ filter }}
    {% endif %}
  {% endcall %}
  {{ return(load_result('collect_freshness')) }}
{% endmacro %}
```

This is also what I assume happens _today_ if a user runs `dbt source freshness` using `dbt-materialize`, since we never explicitly marked source freshness checks as unsupported (despite [this disclaimer](https://github.com/MaterializeInc/materialize/blob/main/misc/dbt-materialize/README.md#sources)).